rfcs/taskforce/0003-Integration-test-infrastructure-spec.md

```diff
@@ -0,0 +1,204 @@
+# Infrastructure Specification for Integration tests
+
+## Goal
+
+To specify the interfaces between different components and the interactions involved in executing an integration test. More concretely, proposing a spec for a Test Executive (pretty close to the one defined [here](https://github.com/CodaProtocol/coda/pull/4715). The role of a test executive is to bring together various components involved in executing an integration test and coordinating the interactions among them.
+These components are:
+    1. Testnet infrastructure: Given a network configuration, deploys a testnet for a specific infrastructure.
+    2. Test: Defining the integration test itself
+    3. Logging: Consuming logs from the deployed testnet for validation
+    4. Automated validation: Defines and executes network-wide validations (Out of scope)
+
+The goal is to have a system where each of these components can be independently implemented and the test executive will be responsible for initializing them and facilitating the interactions.
+
+## Design
+
+### Existing
+
+Integrations tests are currently run on a simulated network using local processes. The test controls and inspects the nodes by communicating over RPC. The tests are tightly coupled to the simulated network and cannot be ported to run on cloud-based or container-based environments. More information on the motivation is described in this [rfc](https://github.com/CodaProtocol/coda/pull/4715) (TODO: Copy it to make it easier to read?)
+
+### Improvements
+
+The proposal for the test executive is part of the whole integration test framework revamp to address the drawbacks of the existing framework (#4735).
+
+### Proposal
+
+#### Infrastructure configuration
+
+#### Test executive
+
+The test executive requires the following inputs to successfully execute an integration test
+
+1. Genesis constants and ledger:
+`Runtime_config.t` currently has all the configurable constants and ledger
+
+A ledger can be specified in any of the following way: (already implemented)
+
+```ocaml
+
+module Accounts = struct
+  type single =
+    { pk: string option [@default None]
+    ; sk: string option [@default None]
+    ; balance: Currency.Balance.t
+    ; delegate: string option [@default None] }
```

Comment:
no timing?

Reply:
Good point, I'll add timing and token_id as optional fields. Any other field that we add to the account in the future can be included here.